### PR TITLE
Add support for Kowl

### LINKED ISSUE #122: Add support for kowl as message viewer
Support switching between AKHQ and [kowl](https://github.com/cloudhut/kowl) in the default linker.

### PR DESCRIPTION
**Pull request by disrupted**
Closes #122

### CODE REVIEW

File: backend/settings.yaml [hunk 1]
```diff
@@ -39,9 +39,14 @@ prometheus:
   url: "http://localhost:9090"
 
 akhq:
+  enable: true
```
Comment (philipp94831) [inline backend/settings.yaml:42]: I think false would actually be best, what do you think @VictorKuenstler ?

File: backend/streams_explorer/defaultlinker.py [hunk 2]
```diff
@@ -13,31 +13,50 @@ def __init__(self):
...
+            )
+            akhq_consumer_link = NodeInfoListItem(
+                name="Consumer Group Details", value="akhq", type=NodeInfoType.LINK
+            )
+            self.add_streaming_app_info_item(akhq_consumer_link)
+            self.add_connector_info_item(akhq_consumer_link)
+
+        if settings.kowl.enable:
+            self.add_topic_info_item(
+                NodeInfoListItem(
```
Comment (philipp94831) [inline backend/streams_explorer/defaultlinker.py:50]: please extract a method as this is almost the same as for akhq

File: backend/streams_explorer/defaultlinker.py [hunk 3]
```diff
@@ -13,31 +13,50 @@ def __init__(self):
...
+                    name="Message Viewer", value="kowl", type=NodeInfoType.LINK
+                )
+            )
+            kowl_consumer_link = NodeInfoListItem(
```
Comment (philipp94831) [inline backend/streams_explorer/defaultlinker.py:54]: see above

**Review by philipp94831 — CHANGES_REQUESTED** [review]

File: backend/streams_explorer/defaultlinker.py [hunk 4]
```diff
@@ -68,8 +71,19 @@ def get_redirect_streaming_app(
...
                 return f"{settings.akhq.url}/ui/{settings.akhq.cluster}/group/{consumer_group}"
-        return None
+            elif link_type == "kowl":
+                return f"{settings.kowl.url}/groups/{consumer_group}"
 
     def get_sink_source_redirects(self, node_type: str, sink_source_name: str):
         if node_type == "elasticsearch-index":
             return settings.esindex.url
+
+    def add_provider(self, name: str):
```
Comment (philipp94831) [inline backend/streams_explorer/defaultlinker.py:81]: please find an a bit more specific method name. Think: provider of *what*?

File: backend/settings.yaml (hunk 1 above)
Comment (VictorKuenstler) [reply backend/settings.yaml:42]: Yes, I like that, then you have to configure the message viewer.

File: backend/settings.yaml [hunk 5]
```diff
@@ -39,9 +39,14 @@ prometheus:
...
   url: "http://localhost:8080"
   cluster: "cluster-name"
 
+kowl:
```
Comment (VictorKuenstler) [inline backend/settings.yaml:46]: Please add a comment, which describes that only either akhq or kowl is supported. Maybe you could even bundle both or the whole linking service configuration into one yaml record:
```yaml
linking:
  message_viewer:
    akhq:
      enable: true
     ...
    kowl:
      enable: false
  ...
  monitoring:
  ...
  logging:
  ...
```
Could also be a separate Issue and PR. Also related to: https://github.com/bakdata/streams-explorer/issues/125
What do you think?

**Review by VictorKuenstler — CHANGES_REQUESTED** [review]

File: backend/settings.yaml (hunk 5 above)
Comment (philipp94831) [reply backend/settings.yaml:46]: Let's restructure the configuration in a later stage along with changes to the helm chart (#123)

Comment (disrupted) [reply backend/settings.yaml:46]: I agree, also I thought that going forward we could actually allow both integrations to be enabled at the same time, we would just need to differentiate it better in the frontend. Adding the service name there would make it clearer imo

File: README.md [hunk 6]
```diff
@@ -152,9 +153,17 @@ The following exporters are required to collect Kafka metrics for Prometheus:
 
 #### AKHQ
 
+- `akhq.enable` Enable AKHQ (bool, **required**, default: `true`)
```
Comment (philipp94831) [inline README.md:156]: update

Comment (disrupted) [reply README.md:156]: touché :D

**Review by philipp94831 — APPROVED** [review]

**Pull request merged**